code.test.js

```diff
@@ -0,0 +1,47 @@
+const fs = require('fs');
+const jsc = require('jsverify');
+
+eval(fs.readFileSync('code.js') + '');
+
+const test = jsc.forall("array (pair nat nat)", function(edges) {
+    let max = edges.reduce((a, b) => Math.max(a, Math.max(b[0], b[1])),0);
+
+    let graph = {};
+    for (let i = 0; i <= max; i++) {
+        graph[i] = [];
+    }
+    
+    for (let [u, v] of edges) {
+        graph[u].push(v);
+    }
+
+    function manaulDetect(graph) {

```

Comment:
This is exactly the same as your implementation, right? You can't check that your implementation is correct by comparing the result to the same code.